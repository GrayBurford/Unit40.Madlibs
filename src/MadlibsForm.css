
.MadlibsForm {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: normal;
    border: 3px black solid;
    border-radius: 10px;
    padding: 5px;
    width: auto;
    margin: 10px;
    padding: 10px;
    color: black;
    font-weight: bolder;
    box-shadow: rgb(38, 57, 77) 0px 20px 30px -10px;
}

input {
    padding: 10px;
    margin: 5px;
    width: 20%;
    border: 1px solid black;
    border-radius: 5px;
}

.MadlibsForm-submit {
    display: flex;
    align-items: normal;
    justify-content: center;
    padding: 10px;
    margin: 10px;
    width: auto;
    border: 1px black solid;
    border-radius: 5px;
    background-color: green;
    color: black;
    font-size: 1rem;
    font-weight: bold;
}


