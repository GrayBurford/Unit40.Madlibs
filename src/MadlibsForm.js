import React, { useState } from 'react';
import './MadlibsForm.css';


// Form component for Madlibs game
// Will generate re-render of the MadlibsBoard component and add new Story component
function MadlibsForm ({ addStory }) {
    const INITIAL_STATE = {
        noun1 : "",
        noun2 : "",
        adjective : "",
        color : ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    function handleSubmit (evt) {
        evt.preventDefault();
        addStory({...formData})
        setFormData(INITIAL_STATE);
    }

    function handleChange (evt) {
        const { name, value } = evt.target;
        setFormData(data => ({...data, [name] : value }))
    }

    return (
        <form className='MadlibsForm' onSubmit={handleSubmit}>
            <div>
                <label htmlFor='noun1'>Noun1</label>
                <input 
                    id='noun1'
                    name='noun1'
                    type='text'
                    placeholder='first noun'
                    value={formData.noun1}
                    onChange={handleChange}
                />
            </div>            
            <div>
                <label htmlFor='noun2'>Noun2</label>
                <input 
                    id='noun2'
                    name='noun2'
                    type='text'
                    placeholder='second noun'
                    value={formData.noun2}
                    onChange={handleChange}
                />
            </div>            
            <div>
                <label htmlFor='adjective'>Adjective</label>
                <input 
                    id='adjective'
                    name='adjective'
                    type='text'
                    placeholder='adjective'
                    value={formData.adjective}
                    onChange={handleChange}
                />                
            </div>
            <div>
                <label htmlFor='color'>Color</label>
                <input 
                    id='color'
                    name='color'
                    type='text'
                    placeholder='color'
                    value={formData.color}
                    onChange={handleChange}
                />                
            </div>
            <button className='MadlibsForm-submit'>Create Madlib!</button>
        </form>
    )
}

export default MadlibsForm;