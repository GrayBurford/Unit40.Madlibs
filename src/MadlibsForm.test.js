import React from "react";
import { render } from "@testing-library/react";
import MadlibsForm from "./MadlibsForm";

// smoke test
test('Renders without crashing', () => {
    render(<MadlibsForm />);
})


// snapshot test
test('Matches snapshot', () => {
    const { asFragment } = render(<MadlibsForm />);
    expect(asFragment()).toMatchSnapshot();
})