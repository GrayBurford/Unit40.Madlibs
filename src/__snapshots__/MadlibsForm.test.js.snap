// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="MadlibsForm"
  >
    <div>
      <label
        for="noun1"
      >
        Noun1
      </label>
      <input
        id="noun1"
        name="noun1"
        placeholder="first noun"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="noun2"
      >
        Noun2
      </label>
      <input
        id="noun2"
        name="noun2"
        placeholder="second noun"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="adjective"
      >
        Adjective
      </label>
      <input
        id="adjective"
        name="adjective"
        placeholder="adjective"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
    </div>
    <button
      class="MadlibsForm-submit"
    >
      Create Madlib!
    </button>
  </form>
</DocumentFragment>
`;
